pkg/cas/hardlinking_content_addressable_storage.go

```diff
@@ -105,6 +110,15 @@ func (cas *hardlinkingContentAddressableStorage) GetFile(ctx context.Context, di
 		if err := directory.Link(name, cas.cacheDirectory, key); err != nil {
 			return err
 		}
+		if cas.deterministicMtimes {
```

Comment:
Instead of fixing this up in the hardlinking cache, would it make more sense to do this as part of bb-storage/pkg/cas/blob_access_content_addressable_storage.go? That's where the files get created in the first place.

Because that function also happens to have a reference to the file descriptor that's used to create the file, we could even decide to use `futimens()` instead of `utimensat()` there.

Reply:
To clarify, you could extend `FileAppender` to gain this function:
```
type FileAppender interface {
        ...
        Chtimes(atime time.Time, mtime time.Time) error
}
```

Then blobAccessContentAddressableStorage.GetFile() could call w.Chtimes() after creating the file has succeeded.

I think that approach is simple and elegant enough that it doesn't necessarily need its own configuration option. Feel free to just enable that unconditionally.